### PR TITLE
Saving tags dialog

### PR DESCRIPTION
**Pull request by BartSoftware**

### CODE REVIEW

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java [hunk 1]
```diff
@@ -113,20 +127,39 @@ boolean extractBundle() {
         return false;
     }
 
+    public EditActivity() {
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java]: I think we don't need this constructor implementation

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java [hunk 2]
```diff
@@ -113,20 +127,39 @@ boolean extractBundle() {
...
+        System.out.println();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        if (!editCompoundRecyclerView.getChangedTags().isEmpty()) {
+            menu.add(0, APPROVE_CHANGES, 0, "Save").setIcon(R.drawable.ic_approve)
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java]: I would like to have "Save" string in resources.

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java [hunk 3]
```diff
@@ -113,20 +127,39 @@ boolean extractBundle() {
...
             default:
                 return super.onOptionsItemSelected(item);
         }
     }
 
-    private void updateTags() {
-        List<Tag> changedTags = editCompoundRecyclerView.getChangedTags();
-        tagsManager.editTags(changedTags);
+    private void startImagesActivity() {
+        Intent intent = new Intent(EditActivity.this, ImagesActivity.class);
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java]: I would move this intent building logic into ImagesActivity static method.

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java [hunk 4]
```diff
@@ -283,14 +316,40 @@ void setEditCompoundRecyclerView(EditCompoundRecyclerView editCompoundRecyclerVi
...
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    void showApproveChangeTagsDialog() {
+        new AlertDialog
+                .Builder(this, R.style.AlertDialogStyle)
+                .setTitle(R.string.tag_changed_title)
+                .setMessage(R.string.tag_changed_message)
+                .setPositiveButton(
+                        getString(android.R.string.yes).toUpperCase(Locale.getDefault()),
+                        (dialog, which) -> {
+                            finish();
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java:342]: We do not see snackbar after saving changes in this case.

**Review by kejmil01 — APPROVED** [review]

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java [hunk 5]
```diff
@@ -1,39 +1,33 @@
...
 
-    @VisibleForTesting
-    public void setSnackbar(Snackbar snackbar) {
-        this.snackbar = snackbar;
+        return intent;
     }
 
     @Override
     public void prepareView() {
+        imagesCallback = builCallback();
```
Comment (michal1990) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesActivity.java]: typo: "buildCallback()", and the method's name should describe which callback you're building e.g. buildImagesCallback()

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesCallback.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import android.support.annotation.VisibleForTesting;
+import android.support.design.widget.Snackbar;
+
+import com.invisibleteam.goinvisible.model.ImageDetails;
+import com.invisibleteam.goinvisible.mvvm.edition.EditActivity;
+
+public class ImagesCallback implements ImagesView {
+
+    private Snackbar snackbar;
+    private Activity activity;
```
Comment (michal1990) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesCallback.java]: it should be final

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesCallback.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    public void navigateToEdit(ImageDetails imageDetails) {
+        Intent intent = EditActivity.buildIntent(activity, imageDetails);
+        activity.startActivity(intent);
+    }
+
+    @Override
+    public void prepareSnackBar(int messageId) {
```
Comment (michal1990) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesCallback.java]: I prefer "messageResId" -> then we know where we can find this message

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesCallback.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        if (snackbar == null) {
+            snackbar = Snackbar
```
Comment (michal1990) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesCallback.java]: this line wrap is not necessary

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesViewModel.java [hunk 9]
```diff
@@ -38,7 +39,9 @@ public void onItemClick(ImageDetails imageDetails) {
 
             @Override
             public void onUnsupportedItemClick() {
-                imagesView.showUnsupportedImageInfo();
+
```
Comment (michal1990) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesViewModel.java]: this extra line is useless

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesView.java [hunk 10]
```diff
@@ -5,5 +5,7 @@
 interface ImagesView {
     void navigateToEdit(ImageDetails imageDetails);
 
-    void showUnsupportedImageInfo();
+    void prepareSnackBar(int messageId);
```
Comment (michal1990) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/images/ImagesView.java]: please add "@StringRes" annotation and change the parameter name to "messageResId"

**Review by michal1990 — APPROVED** [review]

**Pull request merged**